### PR TITLE
Integration test ssh

### PR DESCRIPTION
**Pull request by thepwagner**
Deploy an orbit+app, add random key to SSH db, log in to bastion host.
There's no support for agents, so we can't hop through the bastion to an application instance.

Adding pylint because of default argument fear.
Optimizations: the more you loop deployments the faster you can make them; trimmed ~50s down to ~15s.

### CODE REVIEW

File: src/spacel/provision/alarm/trigger/factory.py [hunk 1]
```diff
@@ -85,7 +85,7 @@ def add_triggers(self, template, triggers, endpoint_resources):
             if dimensions:
                 alarm_properties['Dimensions'] = [
                     {'Name': k, 'Value': v} for k, v in dimensions.items()
-                ]
+                    ]
```
Comment (DeviaVir) [inline src/spacel/provision/alarm/trigger/factory.py]: this looks weird, did pylint enforce this?

File: src/spacel/provision/cloudformation.py [hunk 2]
```diff
@@ -29,10 +31,16 @@ def __init__(self, clients, change_sets, uploader, sleep_time=2):
         self._sleep_time = sleep_time
         self._uploader = uploader
 
-    def _stack(self, name, region, json_template, parameters={},
-               secret_parameters={}):
+    def _stack(self, name, region, json_template, parameters=None,
+               secret_parameters=None):
+        parameters = parameters or {}
+        secret_parameters = secret_parameters or {}
```
Comment (DeviaVir) [inline src/spacel/provision/cloudformation.py]: because python

File: src/spacel/provision/cloudformation.py [hunk 3]
```diff
@@ -29,10 +31,16 @@ def __init__(self, clients, change_sets, uploader, sleep_time=2):
...
         cf = self._clients.cloudformation(region)
-        template_url = self._uploader.upload(json_template, name)
+        template_body = json.dumps(json_template, indent=2, sort_keys=True)
+        if len(template_body) >= 51200:
```
Comment (DeviaVir) [inline src/spacel/provision/cloudformation.py]: nice detail!

File: src/spacel/provision/alarm/trigger/factory.py (hunk 1 above)
Comment (thepwagner) [inline src/spacel/provision/alarm/trigger/factory.py]: I'm guessing my IDE did it, and I just had too big of a diff to notice+revert.

Refactored to something IDE and pylint can agree upon.

**Comment by thepwagner** [comment]
Build failure due to https://github.com/eliben/pycparser/issues/148

There's no rush to merge this; the issue only affects this branch (as `cryptography` is pulled in by the addition of `paramiko` for SSH testing). Shelving to let upstream react+patch and make the eventual fix easier.

**Comment by thepwagner** [comment]
Thanks upstream!

**Pull request merged**